cluster.go

```diff
@@ -193,6 +202,40 @@ func (c *clusterClient) _refresh() (err error) {
 		}
 	}
 
+	shouldRefresh := false
+	c.mu.RLock()
+	// check if the new topology is different from the current one
+	for addr, cc := range conns {
+		old, ok := c.conns[addr]
+		if !ok || old.replica != cc.replica {
+			shouldRefresh = true
+			break
+		}
+	}
+	// check if the current topology is different from the new one
+	if !shouldRefresh {
+		for addr := range c.conns {
+			if _, ok := conns[addr]; !ok {
+				shouldRefresh = true
+				break
+			}
+		}
+	}
+	// check if cluster client is initialized.
+	if !shouldRefresh {
+		for _, cc := range c.pslots {
+			if cc == nil {
+				shouldRefresh = true
+				break
+			}
+		}
+	}
```

Comment:
Is this check really necessary? Also, it is valid that a living cluster has some missing slots.

Reply:
Oh, yes. you are right. 
[9f886c5](https://github.com/redis/rueidis/pull/638/commits/9f886c503d27f212422fdedc97f1b3c050eff45c)